pkg/clusteragent/admission/mutate/auto_instrumentation.go

```diff
@@ -124,13 +195,13 @@ func libImageName(registry string, lang language, tag string) string {
 	return fmt.Sprintf(imageFormat, registry, lang, tag)
 }
 
-func injectAutoInstrumentation(pod *corev1.Pod, _ string, _ dynamic.Interface) error {
+func (apmInstrumentation *APMInstrumentation) injectAutoInstrumentation(pod *corev1.Pod, _ string, _ dynamic.Interface) error {
 	if pod == nil {
 		return errors.New("cannot inject lib into nil pod")
 	}
 	injectApmTelemetryConfig(pod)
 
-	if isApmInstrumentationEnabled(pod.Namespace) {
+	if apmInstrumentation.isEnabled(pod.Namespace) {
```

Comment:
I'm wondering if we could do a function that take the pod as input instead of the namespace string, so we can filter on more field in the futur. 

also maybe the name can be `shouldMutate()` 

```suggestion
	if apmInstrumentation.shouldMutate(pod) {
```

Reply:
I'll leave this like this for now because there's already a `shouldMutatePod` function:

- `isEnabled(namespace)` checks whether apm ssi is enabled in the given namespace.
- `shouldMutate(pod)` checks whether the pod should be mutated. It takes into account its labels, if the mutate unllabeled option is set and if apm ssi is enabled.

We can maybe clean this a bit in the future refactor PR.